components/server/src/github/github-auth-provider.ts

```diff
@@ -98,26 +98,35 @@ export class GitHubAuthProvider extends GenericAuthProvider {
                 .map((s: string) => s.trim())
             );
 
-            const filterPrimaryEmail = (emails: typeof userEmails) => {
+            const blockPassListEmail = ((emails: typeof userEmails) => {
                 if (this.env.blockNewUsers) {
-                    // if there is any verified email with a domain that is in the blockNewUsersPassList then use this email as primary email
+                    // if there is any verified email with a domain that is in the blockNewUsersPassList then use this email as block pass list email
                     const emailDomainInPasslist = (mail: string) => this.env.blockNewUsersPassList.some(e => mail.endsWith(`@${e}`));
                     const result = emails.filter(e => e.verified).filter(e => emailDomainInPasslist(e.email))
                     if (result.length > 0) {
-                        return result[0].email;
+                        return result[0];
                     }
                 }
-                // otherwise use GitHub's primary email as Gitpod's primary email
-                return emails.filter(e => e.primary)[0].email;
-            };
+            })(userEmails);
+            const primary = userEmails.filter(e => e.primary)[0]!;
+            const proxy = userEmails.find(e => e.email.endsWith(`@users.noreply.${this.config.host}`));
+
+            const additionalEmails: { address: string; type: string; }[] = [];
+            if (primary.visibility === 'private' && proxy) {
+                additionalEmails.push({ "address": proxy.email, "type": "commit" });
+            }
+            if (blockPassListEmail) {
+                additionalEmails.push({ "address": blockPassListEmail.email, "type": "block_pass" })
+            }
 
             return <AuthUserSetup>{
                 authUser: {
                     authId: String(id),
                     authName: login,
                     avatarUrl: avatar_url,
                     name,
-                    primaryEmail: filterPrimaryEmail(userEmails)
+                    primaryEmail: primary.email,
+                    additionalEmails,
```

Comment:
I think we should add **all** _verified_ email addresses to `additionalEmails` here. This could also be of use to improve the signup process.

The `proxy` is marked as `type=commit`, which is also available in the GitLab case. 👍🏻 

For the git config we would use `identity.additionalEmails.find(isCommitEmail) || identity.primaryEmail` then.

Reply:
OK, then I would suggest adding all _verified_ and _non-private_ (public) email addresses.

The commit email gets the type `commit` (as it is now) and all others get the type `other`.

Reply:
👍  All verified, non-private, and non-primary emails should now be in the `additionalEmails` array.